### ISSUE TITLE
Implement accessor methods on Holon for descriptor and predecessor 

### ISSUE THREAD

**Issue by evomimic** [evomimic/map-holons#143]
This enhancement replaces the `descriptor` and `predecessor` fields on Holon with accessor functions (get/with). 

For testing reasons, it is recommended this issue be worked in a subbranch of Issue #118.

## Dependencies

- [x] Issue #95 

## Current State

Both of the fields above are, at best, caches of relationships whose source of truth is in the holon's _relationship_map_. The DESCRIBED_BY relationship relates a holon with its type descriptor and the PREDECESSOR relationship relates a holon to its predecessor. 

## Proposal

1. Drop the `descriptor` and `predecessor` fields from Holon.
2. Add `get_descriptor ` and `get_predecessor` methods to HolonReference
3. Add `with_descriptor` and `with_predecessor` methods to StagedReference

### In _holon.rs_

- [x] Drop the `descriptor` and `predecessor` fields from the definition of Holon  

### In holon_reference.rs

- [x] Add `get_descriptor(&self)->Result<Option<HolonReference>, HolonError>`  method.
- [x] Add `get_predecessor(&self)->Result<Option<HolonReference>, HolonError>` method.

**_Implementation Notes:_**

Both of the above should:

- delegate the call to `get_related_holons` method on their variant, passing `CoreSchemaRelationshipTypeName::DescribedBy.to_string()` for DESCRIBED_BY relationship_ name and `CoreSchemaRelationshipTypeName::Predecessor.to_string()` for PREDECESSOR relationship_ name 
- convert the resulting collection to a single HolonReference (or None) and 
- return an Error if more than one `descriptor` or `predecessor` are found.   
- Otherwise, return the HolonReference

### In staged_reference.rs

- [x] Add `with_descriptor(&self, context: &HolonsContext, descriptor_reference: HolonReference)->Result<Self,HolonError>` method for Holon. 
- [x] Add `with_predecessor(&self, context: &HolonsContext, predecessor_reference: HolonReference)->Result<Self,HolonError>` method for Holon. 

**_Implementation Notes:_**

- Both methods should first ensure the Holon _is_accessible_ for Write.
- Currently, Holon's are only allowed to have a single descriptor and (at most) a single predecessor. So these methods will need to first remove any holon related through the relationship in question. 
- Both methods should leverage the implementation of _add_related_holons_ and `remove_related_holons`, but these methods must first be implemented via Issue #95.

## Testing

These methods will NOT be exposed as separate dances as they are really just guest-side convenience functions. Thus, they cannot be tested directly via dance tests. However, they can be tested indirectly as part of Issue #118. 

## Definition of Done

- [ ] This issue may be closed when testing for Issue #118 demonstrates these methods behave correctly.

**Comment by evomimic** [comment]
@dauphin3 hey zeek...  the above design references the CoreSchemaRelationshipTypeName::Predecessor.to_string() name which I later realized I hadn't yet defined.  I added it to PR #146 which I just pulled into main. You can grab crates/coordinator/core_schema/src/relationship_type_loader.rs from `main` to give you access to that name in your branch.

**Comment by evomimic** [comment]
@dauphin3 : I intend to resolve the circular dependency problem in a separate issue. Please just use the hard-code relationship names in #143 to bypass the issue and unblock your progress towards completing this issue.

**Comment by evomimic** [comment]
Now that #118 has been merged into main, this issue can be closed.
